src/core/components/collapsible/collapsible.tsx

```diff
@@ -0,0 +1,75 @@
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { Button } from '../button';
+import { Icon, IconType } from '../icon';
+import { type ICollapsibleProps } from './collapsible.api';
+
+const collapsedSizeVariants = {
+    sm: 'h-32',
+    md: 'h-64',
+    lg: 'h-96',
+};
+
+export const Collapsible: React.FC<ICollapsibleProps> = ({
+    collapsedSize = 'md',
+    customCollapsedSize,
+    defaultOpen = false,
+    buttonLabelOpened,
+    buttonLabelClosed,
+    buttonVariant,
+    className,
+    blinderClassName,
+    onToggle,
+    children,
+    ...otherProps
+}) => {
+    const [isOpen, setIsOpen] = useState(defaultOpen);
+
+    const toggle = useCallback(() => {
+        setIsOpen((prev) => {
+            const newState = !prev;
+            onToggle?.(newState);
+            return newState;
+        });
+    }, [onToggle]);
+
+    const contentClassNames = classNames(
+        'mb-4 overflow-hidden ease-in-out md:mb-6',
+        isOpen ? 'h-auto' : collapsedSizeVariants[collapsedSize],
```

Comment:
The variants will lead to an issue where if your content is not enough to fill up 'lg' for example, the content will still grow to 'lg', pushing the trigger to the very bottom with only whitespace. I'm not sure about that approach as opposed to one where the collapsedSize is dynamically calculated based on the size of the content (or honestly even more straightforward, clamping the content.)